### PR TITLE
Server Setup

### PR DESCRIPTION
**Pull request by alexapostolu**
Created the Nodejs server setup.

* Wrote Routes for **Challenge**, **Comment**, **Solution**, and **User**.
* Wrote function definitions for the corresponding Controllers.

### CODE REVIEW

File: server/db.js
Comment (arthurhohinng) [inline server/db.js:1]: We don't need this file right? As we are using sequelize

File: server/app.js
Comment (arthurhohinng) [inline server/app.js:1]: Seems like there is some overlap in this PR and this https://github.com/utmgdsc/UML_Mentor/pull/11/files#diff-bcb729747c92e72c12e4590f736a334041a572c429756da8af60b6ab89c6be79

May need to connect with Omar and resolve the conflicts

File: server/routes/userRoutes.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+const router = express.Router();
+const UserController = require('../controllers/userController');
```
Comment (Vladmidir) [inline server/routes/userRoutes.js:3]: where the controllers at ?

**Comment by alexapostolu** [comment]
I fixed the issues pointing out in the reviews, and made sure my code has zero merge conflicts with Omar's `database` branch.

**Pull request merged**